<a href="https://colab.research.google.com/github/YasminGarcia1210/MAESTRIA_DATOS/blob/main/Trabajo_Final_DE_Analisis_Exploratorio_Yasmin.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
#Inicamos la lectura desde la fila 8 donde se encuentran los encabezados de las filas
import pandas as pd
import matplotlib.pyplot as plt
datos=pd.read_csv('UsuariosActivos.csv',skiprows=8,delimiter=';')

In [ ]:
datos['SERVICIO'] = pd.to_numeric(datos['SERVICIO'], errors='coerce')

# Elimina las filas que no se pudieron convertir en números y deja solo en contenido número que representa la identificación del servicio
datos = datos.dropna(subset=['SERVICIO'])

In [ ]:
data = datos[['SERVICIO','SUSCRIPC','CATE','SUBC','BARRIO','DESCRIPCION','CUENTAS','SALDO','V.CUENTA','V.IVA','FECH INST']].copy()

data = data.sort_values(by='SUSCRIPC', ascending=False)
data = data.reset_index(drop=True)

# 3. Análisis exploratorio





### 3.1 Análisis de la estructura

#### 3.1.1 Tamaño de la data

In [ ]:
data.shape

(15191, 11)

Los datos cuentan con 15191 observaciones y 11 columnas, esto despues de limpiar la data con columnas que no se necesitan para este análisis

 #### 3.1.2 Datos Nan, Nul

In [ ]:
data.isna().sum()

SERVICIO       0
SUSCRIPC       0
CATE           0
SUBC           0
BARRIO         2
DESCRIPCION    2
CUENTAS        2
SALDO          2
V.CUENTA       2
V.IVA          2
FECH INST      5
dtype: int64

In [ ]:
data.isnull().sum()

SERVICIO       0
SUSCRIPC       0
CATE           0
SUBC           0
BARRIO         2
DESCRIPCION    2
CUENTAS        2
SALDO          2
V.CUENTA       2
V.IVA          2
FECH INST      5
dtype: int64

Podemos observar que los datos Nan y Null coinciden en las columnas, por lo que tomaremos la decisión de borrar los datos Nan=Nul


In [ ]:
data=data.dropna()
data = data.reset_index(drop=True)

In [ ]:
# Quedando asi

data.isna().sum()

Se muestra a continuación las columnas con las que vamos a trabajar

In [ ]:
list(data.columns)

['SERVICIO',
 'SUSCRIPC',
 'CATE',
 'SUBC',
 'BARRIO',
 'DESCRIPCION',
 'CUENTAS',
 'SALDO',
 'V.CUENTA',
 'V.IVA',
 'FECH INST']

### 3.2 Análisis de Variables categoricas

Segun el campo de descripción, las variables categóricas son:

In [ ]:
data.dtypes

SERVICIO       float64
SUSCRIPC        object
CATE            object
SUBC            object
BARRIO          object
DESCRIPCION     object
CUENTAS         object
SALDO          float64
V.CUENTA       float64
V.IVA          float64
FECH INST       object
dtype: object

Sin embargo, las columnas SALDO, CUENTA, U.IVA y FECHA no son variables categoricas sino numéricas, por lo que vamos a hacer la transformación.

In [ ]:
data.head(2)

SERVICIO SUSCRIPC CATE SUBC           BARRIO  \
0  7801959.0    99996    8    2  BELLO HORIZONTE   
1  6801251.0    99996   60   62  BELLO HORIZONTE   

                     DESCRIPCION CUENTAS  SALDO  V.CUENTA   V.IVA FECH INST  
0   INTERNET GPON 40 MB OCT-2022       0    0.0   42680.0 -1153.3   9/08/18  
1  DOBLE INT+TV GPON TV OCT-2022       0    0.0   20940.0  3344.0   6/05/23

In [ ]:
data.dtypes

SERVICIO       float64
SUSCRIPC        object
CATE            object
SUBC            object
BARRIO          object
DESCRIPCION     object
CUENTAS         object
SALDO          float64
V.CUENTA       float64
V.IVA          float64
FECH INST       object
dtype: object

In [ ]:
data['V.CUENTA'] = data['V.CUENTA'].str.replace(',', '').astype(float)
data['SALDO'] = data['SALDO'].str.replace(',', '').astype(float)
data['V.IVA'] = data['V.IVA'].str.replace(',', '').astype(float)